Source/Core/Core/Config/GraphicsSettings.cpp

```diff
@@ -4,13 +4,48 @@
 
 #include "Core/Config/GraphicsSettings.h"
 
+#include <optional>
 #include <string>
 
+#include "Common/StringUtil.h"
 #include "Core/Config/Config.h"
 #include "VideoCommon/VideoConfig.h"
 
 namespace Config
 {
+std::optional<int> ConvertFromLegacyEFBScale(int efb_scale)
+{
+  // In game INIs, -1 was used as a special value meaning
+  // "use the value from the base layer but round it to an integer scale".
+  // We only support integer scales nowadays, so we can simply ignore -1
+  // in game INIs in order to automatically use a previous layer's value.
+  if (efb_scale < 0)
+    return {};
+
+  return efb_scale - (efb_scale > 0) - (efb_scale > 2) - (efb_scale > 4);
```

Comment:
Choice | Old numbering | New numbering
------ | -------------------- | --------------------
Auto (fractional) | 0 | 
Auto (integral) | 1 | 0
1x | 2 | 1
1.5x | 3 | 
2x | 4 | 2
2.5x | 5 | 
3x | 6 | 3
4x | 7 | 4

Basically, for each choice that the new numbering skips, the difference between the two formats increases by one.